docs/design/satellite-svc-separation.md

```diff
@@ -0,0 +1,198 @@
+
+For the metainfo loop and observer system its not critical to have high availability for these systems, therefore its ok to have all these observers depend on a single metainfo loop process.  In the worst case scenario, if the metainfo loop goes down there may be downtime until its fixed and back up, this should be fine for small periods of time.
+
+#### satellite api
+
+We chose to combine all public GRPC and HTTP endpoints into one binary. The reason for this is so that we handle all requests in one place so we can scale handling requests up easier.
+
+## Implementation
+
+Note: getting Kademlia removed is a blocker to implementation.
+
+We should break out one process at a time from the Satellite.  Here are the things we should do first since they impedes scale the most:
+- satellite api
+- repair workers
+- audit workers
+
+We will need to add a `SatelliteSystem` in `testplanet` so that we can test continue using the same unit tests for the Satellite as we break it apart. See an example of that [here](https://github.com/storj/storj/pull/2836/files#diff-c2ce7a2c9b2f4920e2de524c0fffd2f1R70).
+
+For each new satellite binary we need to do the following steps:
+- create xProcess, where x is a Satellite service, e.g. RepairProcess, AccountingProcess, etc
+- update satellite binary to include a subcommand so that the new xProcess can be run like this:
+`satellite run repair`
+- look for any areas in the code that prevent horizontal scaling for this new process and fix if found
+- update testplanet so that unit tests still pass
+- update storj-sim so that integration tests still pass
+- create kubernetes deployment configs (this includes a dockerfile and an k8s HPA resource)
+- automated deployment to staging kubernetes environment is setup and deploying
+
+There is a prototype PR with an example that implements these steps (minus deployment configs) for the repair service, see that [here](https://github.com/storj/storj/pull/2836). This is now out of date since we are adding the repair (checker) observer to the metainfo loop observer system, but this prototype is still useful as an example.
+
+Other work:
+- Add support for a cache for the live accounting cache so that its not stored in memory any longer.
+- add support for postgres to revocationDB.
+- Update storj-sim to run with all the new services.
+
+## Open issues
+- with kademlia/discovery/bootstrap being removed, what will be taking the place of the functionality still needed? For example, what will handle the storage nodes pinging satellite to ensure uptime checks?
+- is uptime system going to be a thing?
+- will we need to run more than one replica of the metainfo loop and its observer system? if so, there are two areas that may prevent scaling: 1) the repair.checkerObserver.ReliabilityCache and 2) the GC observer storing bloom filters in-memory.
```

Comment:
It seems safe to say that we don't need to design for multiple concurrent metainfo loops right now. If the need for that arises, it will be down the road and we'll know a lot more about the specifics of the need at that time.